CleanWeather/Scenes/FavouriteCities/FavouriteCitiesWorker.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  FavouriteCitiesWorker.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 05/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import Foundation
+
+typealias FetchFavouriteCitiesCompletion = (Result<[City], Error>) -> Void
+
+protocol FavouriteCitiesWorker {
+    func fetchAllCities(completion: FetchCitiesCompletion?)
+    func fetchFavouriteCities(completion: FetchFavouriteCitiesCompletion?)
+    func toogleFavourite(for city: City)
+}
+
+final class FavouriteCitiesWorkerImpl {
+
+    private let networking: WeatherNetworking
+    private let database: FavouriteCityRepository
+
+    init(networking: WeatherNetworking, database: FavouriteCityRepository) {
+        self.networking = networking
+        self.database = database
+    }
+}
+
+extension FavouriteCitiesWorkerImpl: FavouriteCitiesWorker {
+
+    func fetchAllCities(completion: FetchCitiesCompletion?) {
+        networking.fetchCities(completion: completion)
+    }
+
+    func fetchFavouriteCities(completion: FetchFavouriteCitiesCompletion?) {
+        let favouriteCities = database.getFavouriteCities()
+        completion?(.success(favouriteCities))
```

Comment:
i want here:

database - rename it to repository
repository.fetchFavouriteCitires(completion: completion)